src/plugin.ts

```diff
@@ -0,0 +1,33 @@
+import { Web3PluginBase, utils } from "web3";
+import { AddEthereumChainRequest } from "./types";
+
+type WalletRpcApi = {
+  wallet_addEthereumChain: (param: AddEthereumChainRequest) => null;
+};
+
+export class WalletRpcPlugin extends Web3PluginBase<WalletRpcApi> {
+  public pluginNamespace = "walletRpc";
+
+  public constructor() {
+    super();
+  }
+
+  public async addEthereumChain(param: AddEthereumChainRequest): Promise<null> {
```

Comment:
Documentation comments here would be nice

Reply:
This is coming in the next PR